src/v/s3/client.cc

```diff
@@ -606,14 +606,21 @@ ss::future<> client::put_object(
     auto header = _requestor.make_unsigned_put_object_request(
       name, id, payload_size, tags);
     if (!header) {
-        return ss::make_exception_future<>(std::system_error(header.error()));
+        return body.close().then([&header] {
+            return ss::make_exception_future<>(
+              std::system_error(header.error()));
```

Comment:
this looks like a use-after-free violation. the continuation attached to `body.close()` captures a reference to a variable on the stack (`header`) and then accesses that variable after returning from the function at which point `header` will have been destroyed.